docs/content/docker.md

```diff
@@ -0,0 +1,280 @@
+part is so that the plugin can create a `json` file to know what volumes
+existed in case of a system restart, etc.
+
+When the container is destroyed, etc, it will look at a file it created in
+`/var/lib/docker/plugins/rclone/docker-plugin.state` to recreate any volumes
+that had previously existed, so you want that JSON file to persist on the host.
+
+## Creating Volumes
+
+You can create a volume with this plugin:
+```
+docker volume list
+docker volume create vol1 -d rclone -o remote=storj: -o vfs-cache-mode=full
+docker volume create vol2 -d rclone -o remote=:tardigrade,access_grant=xxx:heimdall
+docker volume create vol3 -d rclone -o type=tardigrade -o path=heimdall -o tardigrade-access-grant=xxx -o poll-interval=0
+docker volume inspect vol1
+```
+
+When you create a container, you can connect it to the Volume:
+```
+docker run --rm -it -v vol1:/mnt --workdir /mnt ubuntu:bionic
+```
+
+Rclone flags and volume options are set through the `-o` flag to the
+`docker volume create` command. There are 5 special options:
+- "remote" aka "fs" determines existing remote from config file
+  with a path or on-the-fly remote using the ":backend:" syntax.
+  It is usually named "remote" in documentation but can be aliased as
+  "fs" to avoid confusion with the "remote" option of some backends.
+- "type" is equivalent to the ":backend:" syntax (optional).
+- "path" provides explicit on-remote path for "type" (optional).
+- "mount-type" can be "mount", "cmount" or "mount2",
+  defaults to first found (optional).
+- "persist" is reserved for future to create remotes persisted
+  in rclone.conf similar to rcd (optional).
+
+Unlike `rclone rcd` we use flat naming scheme for mount, VFS and backend
+options without substructures. Dashes, underscores and mixed case in option
+names can be used interchangeably. Option name conflicts can be resolved
+in a manner similar to rclone CLI by adding prefixes:
+"vfs-", primary mount backend type like "sftp-", and so on.
+After triaging the options are put in MountOpt, VFSOpt or connect
+string for actual filesystem setup and in volume.Options for saving
+the state.
+
+Please note that you can provide options only for the backend immediately
+referenced by the mounted remote or remote type.
+If this is a wrapping backend like alias, chunker or crypt, you cannot provide
+options for the underlying remote/backend.
+The only way to deal with it is to pass `rclone.conf` to the plugin or set
+plugin args. This limitation is imposed by the connection string parser,
+"serve docker" just follows the suit.
+There is however a syntactic difference. Inside connection string the backend
+prefix must be dropped but in the --opt=x array it must be present. This is so
+because --opt=x can include mount/vfs flags now and possibly other parameters
+in future. Also it helps to discriminate between "remote" filesystem path and
+"crypt-remote" backend setting.
+
+## Using with Docker Compose
```

Comment:
This part seems to be a copy-paste not updated. 
Maybe this is not needed since the version 3 example can provide enough information.

Reply:
I looked again at `docker.md`. It has yet more confusing copy-paste places and needs even more cleanup. Sorry :(

Reply:
The docker-compose doc section is similar to the swarm section because docker guys architected swarm config format from the compose one. However, the two modes are different by nature. Compose runs on a single node creating standalone volumes. Swarm volumes are cloned across nodes under control of swarm manager. Their descriptions after initial parsing are stored in swarm log shared and negotiated by swarm masters. I will think a little now, then remove copy-paste artifacts and keep only relevant stuff.